test/RL/representation/default/defaulttrajectorystate.jl

```diff
@@ -0,0 +1,67 @@
+@testset "defaulttrajectorystate.jl" begin
+    
+    @testset "DefaultTrajectoryState" begin
+        graph = Matrix(adjacency_matrix(random_regular_graph(6,3)))
+        ts1 = SeaPearl.DefaultTrajectoryState(
+            GeometricFlux.FeaturedGraph(graph; nf=rand(3,6)),
+            2,
+            collect(3:6)
+        )
+        ts2 = SeaPearl.DefaultTrajectoryState(
+            GeometricFlux.FeaturedGraph(graph; nf=rand(3,6)),
+            2
+        )
+
+        batched1 = ts1 |> cpu
+        batched2 = ts2 |> cpu
+        @test isa(batched1, SeaPearl.BatchedDefaultTrajectoryState{Float32})
+
+        @test all(ts1.fg.graph .== batched1.fg.graph[:, :, 1])
+        @test all(abs.(ts1.fg.nf .- batched1.fg.nf[:,:,1]) .< 1e-5)
```

Comment:
Where does the approximation comes from ?

Reply:
We are forced to work with `Float32` with the batched version (for GPU compatibility, which are very unefficient on `Float64`). But the iterative one can use any datatype, and by default Julia takes `Float64` thus inducing approximation errors.